SofaKernel/framework/sofa/core/Mapping.h

```diff
@@ -423,19 +428,19 @@ extern template class SOFA_CORE_API Mapping< sofa::defaulttype::Rigid3fTypes, so
 extern template class SOFA_CORE_API Mapping< sofa::defaulttype::Vec3fTypes, sofa::defaulttype::Rigid3dTypes >;
 
 // outside of Sofa templates
-extern template class SOFA_CORE_API Mapping< sofa::defaulttype::Vec2dTypes, sofa::defaulttype::ExtVec2fTypes >;
+extern template class SOFA_CORE_API Mapping< sofa::defaulttype::Rigid3fTypes, sofa::defaulttype::ExtVec3dTypes >;
 extern template class SOFA_CORE_API Mapping< sofa::defaulttype::Vec2fTypes, sofa::defaulttype::ExtVec2dTypes >;
-
-extern template class SOFA_CORE_API Mapping< sofa::defaulttype::Vec3dTypes, sofa::defaulttype::ExtVec3fTypes >;
 extern template class SOFA_CORE_API Mapping< sofa::defaulttype::Vec3fTypes, sofa::defaulttype::ExtVec3dTypes >;
-
-extern template class SOFA_CORE_API Mapping< sofa::defaulttype::Vec6dTypes, sofa::defaulttype::ExtVec3fTypes >;
 extern template class SOFA_CORE_API Mapping< sofa::defaulttype::Vec6fTypes, sofa::defaulttype::ExtVec3dTypes >;
 
+#endif //SOFA_WITH_FLOAT
+
+// this is needed even if SOFA_DOUBLE only is compiled
+extern template class SOFA_CORE_API Mapping< sofa::defaulttype::Vec2dTypes, sofa::defaulttype::ExtVec2fTypes >;
+extern template class SOFA_CORE_API Mapping< sofa::defaulttype::Vec3dTypes, sofa::defaulttype::ExtVec3fTypes >;
+extern template class SOFA_CORE_API Mapping< sofa::defaulttype::Vec6dTypes, sofa::defaulttype::ExtVec3fTypes >;
```

Comment:
If SOFA_FLOATING_POINT_TYPE=double, then SOFA_WITH_DOUBLE is defined so line 341 is activated.
Why this block?

Reply:
I think this block should be just before the closing `#endif //SOFA_WITH_FLOAT` line 436 and the similar block line 341 should be deleted.

Reply:
Sorry, that was a duplicated bloc that I didn't remove after moving it in SOFA_WITH_DOUBLE:
VecXd -> ExtVecXf are required both when compiling in SOFA_WITH_DOUBLE and in SOFA_WITH_DOUBLE + SOFA_WITH_FLOAT. So technically, they could either be in the  SOFA_WITH_DOUBLE bloc or in the combination of both, it would be the same.
Stupid logic. can't wait to see it dissapear...